### PR TITLE
Dockerfile and cache in ci

### PR DESCRIPTION
**Pull request by webbdays**

### CODE REVIEW

**Comment by webbdays** [comment]
#12

File: Dockerfile-linux [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+COPY . ./
+
+# build typescript to javascript
+RUN npm run build
+
+#global installation of mizu
+RUN npm install -g
+
+# check if mizu is installed and working fine
+RUN mizu --version
```
Comment (solipsis-project) [inline Dockerfile-linux:20]: This doesn't need to be in the Dockerfile. But we do want to set mizu as the default entrypoint (with the CMD action.)

File: .github/workflows/integration-tests.yaml [hunk 2]
```diff
@@ -12,11 +12,47 @@ jobs:
...
     runs-on: ${{ matrix.os }}
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-node@v3
+      - name: Checkout the current branch of mizu-client repo here.
+        uses: actions/checkout@v3
+
+
+      - name: Set up Docker Buildx
+        if: ${{ matrix.os == 'ubuntu-latest' }}
```
Comment (solipsis-project) [inline .github/workflows/integration-tests.yaml:20]: What is the thought process behind only testing Docker and linux and only testing non-docker deployment on Windows?

File: .github/workflows/integration-tests.yaml [hunk 3]
```diff
@@ -12,11 +12,47 @@ jobs:
...
+        uses: docker/setup-buildx-action@v2
+      # - name: Login to DockerHub (we can even login to other container registries)
+      #   if: ${{ matrix.os == 'ubuntu-latest' }}
+      #   uses: docker/login-action@v2
+      #   with:
+      #     username: ${{ secrets.DOCKERHUB_USERNAME }}
+      #     password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Build and push
```
Comment (solipsis-project) [inline .github/workflows/integration-tests.yaml:28]: If we're not currently pushing to DockerHub, we should choose a more accurate name for this build step.

File: Dockerfile-linux (hunk 1 above)
Comment (webbdays) [reply Dockerfile-linux:20]: Ha. yes not needed.
But just there to test mizu was successfully installed in that docker image.
if "RUN mizu --version" fails then docker build fails which is desirable.

File: .github/workflows/integration-tests.yaml (hunk 2 above)
Comment (webbdays) [reply .github/workflows/integration-tests.yaml:20]: currently, i am not able to find a way to build windows container image with docker.
So, just went with docker image build only on linux for now.
Let me know if there is a way to build windows container images with docker or any other builder.

File: .github/workflows/integration-tests.yaml (hunk 3 above)
Comment (webbdays) [reply .github/workflows/integration-tests.yaml:28]: Ha, yes need to be a common name like "Login to container registry"
I was unsure which container registry you prefer/use. So i used dockerhub and commented for time being.

File: Dockerfile-linux (hunk 1 above)
Comment (webbdays) [reply Dockerfile-linux:20]: I will push changes for mizu as the default entrypoint.

Comment (solipsis-project) [reply Dockerfile-linux:20]: "if "RUN mizu --version" fails then docker build fails which is desirable."

Ah, good point. I didn't consider that.

File: .github/workflows/integration-tests.yaml (hunk 3 above)
Comment (solipsis-project) [reply .github/workflows/integration-tests.yaml:28]: Thanks. Leaving it commented out is a good idea, will make it easier to add a container registry later.

File: .github/workflows/integration-tests.yaml (hunk 2 above)
Comment (solipsis-project) [reply .github/workflows/integration-tests.yaml:20]: My understanding is that the os field here is for the host machine, not the docker image. 

I think we want to use the integration tests to verify that Mizu runs on both windows and linux, and that the docker file works. And thinking about it, the way you have it set up here accomplishes all of those goals.

File: .github/workflows/integration-tests.yaml (hunk 3 above)
Comment (webbdays) [reply .github/workflows/integration-tests.yaml]: updated build step name to Build

File: Dockerfile-linux (hunk 1 above)
Comment (webbdays) [reply Dockerfile-linux]: added default command
CMD ["mizu", "--help"]

Comment (webbdays) [reply Dockerfile-linux]: If the container is supposed to be used in the below steps of workflow for some tests then it wont make sense to add CMD ["mizu", "--help"] .
But this command make sense when there is some real human using the built container. (But we are not saving docker image here)

**Review by solipsis-project — APPROVED** [review]

**Comment by webbdays** [comment]
Can you please help me in signing all my previous commits in this branch (dockerfile).

**Comment by solipsis-project** [comment]
I ove

**Comment by solipsis-project** [comment]
Don't worry about signing the commits; we're going to squash them all for the pull. I manually overrode needing the commits to be signed.

**Comment by solipsis-project** [comment]
Thank you for taking on this issue and submitting the pull request!

**Pull request merged**